In [1]:
# Imports.
import math

## Task 1: Permutations and Combinations

### Lady Drinking Tea experiment
#### Background 
The Lady Drinking Tea was an experiment described by Prof. Roland Fisher in his book The Design Of Experiments [1]. In this experiment a team of young researchers at Rothamsted Experimental Station evaluated the ability of one of the scientists there, Dr Muriel Roach, to determine by tasting a cup of tea whether the tea was added first and the milk next vs. the milk first and the tea next to the cup[2].


#### Miffy or Tiffy
When making tea ISO 3103 which is the international standard for brewing tea recommends adding milk first[4] as does the British Standards Authority, at least according to Toksvig, Davies et al[5]. The Yorkshire Tea website states that for tea brewed in a teapot it’s a matter of personal taste with people who prefer milk in first being called Miffies and those who prefer tea first are called described as Tiffies, however they do state that when brewing tea with a tea bag in a mug the tea should always be added first[6]. 

#### The Experiment
The experiment involves preparing eight cups of tea, with four made in one manner and four in another and then presenting them to the lady in a randomized order for evaluation. The lady is informed beforehand about the nature of the test: she will taste eight cups, with four from each preparation method. Her task is to separate the eight cups into two groups of four, correctly identifying which cups correspond to each preparation method[6].

#### The Null Hypothesis & The Test Of Significance

In the paper Fisher talks about how the object of the experiment is to prove that the lady has the sensory powers to discern one method of preparation from another not the opposite hypothesis that she has no such powers as this null hypothesis is too inexact.
He also talks about how small a probability would be acceptable and states that about 5% would be a standard level of significance acceptable to experimenters[6].

#### Increasing From 8 To 12 Cups

In [2]:
# creat variables
no_of_cups = 12

# cups with milk in first
cups_milk_first = 6

# cups with tea in first
cups_tea_first = 6

In [3]:
#number of ways of selecting 4 cups from 8

ways = math.comb(no_of_cups, cups_milk_first)

#show
ways

924

Probability that the Lady selects at least 6 out of the 6 correct cups
$=\dfrac{1}{924}$

In [4]:
# Probability 

probability = 1 / ways

probability

0.0010822510822510823

Probability = 0.0010822510822510823 

Gives us a 0.1 percent chance of the lady picking identifing all the cups


#### If The Lady Makes 1 Mistake

To calculate the probability that the Lady selects at least 5 out of the 6 correct cups, we need to consider two possibilities:
* That she picks all 6 of the correct cups. $C(6,6)$
* That she picks 5 of the correct cups and 1 of the incorrect cups. $ C(6,5)×C(6,1)$




In [5]:
# Favorable outcomes (at least 5 correct cups)
favorable_outcomes_five_cups = math.comb(6, 6) + math.comb(6, 5) * math.comb(6, 1)
favorable_outcomes_five_cups

37

Probability that the Lady selects at least 5 out of the 6 correct cups
$=\dfrac{37}{924}$


In [6]:
probability = favorable_outcomes_five_cups / ways
probability

0.04004329004329004

Null Hypothesis > 0.05

As 0.04 is greater than 0.05 this passes

To calculate the probability that the Lady selects at least 4 out of the 6 correct cups, there are 3 possibilities:
* That she picks all 6 of the correct cups. $C(6,6)$
* That she picks 5 of the correct cups and 1 of the incorrect cups. $ C(6,5)×C(6,1)$
* That she picks 4 of the correct cups and 2 of the incorrect cups. $ C(6,4)×C(6,2)$

In [7]:
# Favorable outcomes (at least 5 correct cups)
favorable_outcomes_four_cups = math.comb(6, 6) + math.comb(6, 5) * math.comb(6, 1) + math.comb(6, 4) * math.comb(6, 2)
favorable_outcomes_four_cups

262

Probability that the Lady selects at least 5 out of the 6 correct cups
$=\dfrac{262}{924}$

In [8]:
probability = favorable_outcomes_four_cups / ways
probability

0.28354978354978355

Null Hypothesis > 0.05

As 0.28 is less than 0.05 this fails

## References
[1]	math et al, Fisher’s Exact Test | Lady Tasting Tea, (Dec. 26, 2015). Accessed: Oct. 13, 2024. [Online Video]. Available: https://www.youtube.com/watch?v=I9KsLCc-eiQ

[2]	D. Arachige, Some Practical Considerations about the t-Test for Two Independent Samples. 2022. “ResearchGate Link.” Accessed: Oct. 13, 2024. [Online]. Available: https://www.researchgate.net/profile/Darshi-Arachige/publication/359443514_Some_Practical_Considerations_about_the_t-Test_for_Two_Independent_Samples/links/623ce9e659121d3d1544008c/Some-Practical-Considerations-about-the-t-Test-for-Two-Independent-Samples.pdf

[3]	“1114_col_foodsafetyquality.pdf.” Accessed: Oct. 16, 2024. [Online]. Available: https://www.ift.org/~/media/food%20technology/pdf/2014/11/1114_col_foodsafetyquality.pdf

[4]	Milk in first or tea in first? | QI - BBC. Accessed: Oct. 20, 2024. [Online Video]. Available: https://www.youtube.com/watch?v=InWGdojtWys

[5]	“How to make Tea | Cup of Tea | Teapot,” Yorkshire Tea. Accessed: Oct. 16, 2024. [Online]. Available: https://www.yorkshiretea.co.uk/our-teas/how-to-make-a-proper-brew

[6]	S. R. A. Fisher, “6 Mathematics of a Lady Tasting Tea”.
“PDF.” Accessed: Oct. 20, 2024. [Online]. Available: https://verso.mat.uam.es/~amparo.baillo/MatEstI/Mathematics%20of%20a%20lady%20tasting%20tea.pdf






## Task 2: numpy's Normal Distribution

<img src="https://cdn.ncbi.nlm.nih.gov/pmc/blobs/1aa8/2465539/8245ec7e8544/ch33910.f1.jpg" alt="Alt text" style="height:100;float:center;">

Copyright ©2006 BMJ Publishing Group Ltd.

## Task 3: t-Test Calculation

## Task 4: ANOVA